packages/realm/src/app-services/App.ts

```diff
@@ -131,13 +134,20 @@ export class App<FunctionsFactoryType = DefaultFunctionsFactory, CustomDataType
 
   /** @internal */
   public static getAppByUser(userInternal: binding.SyncUser): App {
-    const app = this.appByUserId.get(userInternal.identity)?.deref();
+    const app = App.appByUserId.get(userInternal.identity)?.deref();
     if (!app) {
       throw new Error(`Cannot determine which app is associated with user (id = ${userInternal.identity})`);
     }
     return app;
   }
 
+  /** @internal */
+  public static setAppByUser(userInternal: binding.SyncUser, currentApp: AnyApp): void {
+    if (currentApp) {
```

Comment:
I can't imagine this ever being falsy?

Reply:
The check could probably be removed since it's an internal method. I think it just got added since that code was previously used where the `currentApp` as optional 👍